src/debugger/gdb-stub.c

```diff
@@ -29,6 +29,26 @@ enum {
 	MACH_O_ARM_V4T = 5
 };
 
+const char* TARGET_XML = "<target version=\"1.0\">"
+                         "<architecture>armv4t</architecture>"
+                         "<osabi>none</osabi>"
+                         "</target>";
+
+const char* MEMORY_MAP_XML = "<memory-map version=\"1.0\">"
+                             "<memory type=\"rom\" start=\"0x00000000\" length=\"0x00004000\"/>"
+                             "<memory type=\"ram\" start=\"0x02000000\" length=\"0x00040000\"/>"
+                             "<memory type=\"ram\" start=\"0x03000000\" length=\"0x00008000\"/>"
+                             "<memory type=\"ram\" start=\"0x04000000\" length=\"0x00000400\"/>"
+                             "<memory type=\"ram\" start=\"0x05000000\" length=\"0x00000400\"/>"
+                             "<memory type=\"ram\" start=\"0x06000000\" length=\"0x00018000\"/>"
+                             "<memory type=\"ram\" start=\"0x07000000\" length=\"0x00000400\"/>"
+                             "<memory type=\"ram\" start=\"0x08000000\" length=\"0x02000000\"/>"
+                             "<memory type=\"ram\" start=\"0x0A000000\" length=\"0x02000000\"/>"
+                             "<memory type=\"ram\" start=\"0x0C000000\" length=\"0x02000000\"/>"
+                             "<memory type=\"ram\" start=\"0x0E000000\" length=\"0x00020000\"/>"
+                             "<memory type=\"ram\" start=\"0x0F000000\" length=\"0x00020000\"/>"
+                             "</memory-map>";
```

Comment:
This needs to be dynamically generated, as the SRAM region can change. As I mentioned, you should be using core->listMemoryBlocks to do this. Further, you have ROM listed as ram, for...some reason.

Reply:
I asked in the discord and someone told me that the first 0x4000 would be read only, but that the rest are writable.

rom just means it's read only and ram means read/write. If it's wrong, then which regions should be marked as rom? the 3 cart one + bios? are there others?

Reply:
The API in question contains this information...

Also, the ROM is read-only except in very specific circumstances: there is a small region reserved for GPIO access, but that only exists in a handful of games, and there is another small (but not as small) region which is used for EEPROM in some games. Beyond that, it cannot be modified except by internal API (the rawWrite* calls, specifically).